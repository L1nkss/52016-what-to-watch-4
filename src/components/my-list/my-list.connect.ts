import MyList from "@components/my-list/my-list";
import {connect} from "react-redux";
import {getFavoriteFilms} from "@components/my-list/selectors/selectors";

const mapStateToProps = (state) => ({
  favoriteFilms: getFavoriteFilms(state)
});

export default connect(mapStateToProps, null)(MyList);
