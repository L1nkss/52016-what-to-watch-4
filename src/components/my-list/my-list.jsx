import Header from "@components/header/header.connect";
import FilmCardItem from "@components/film-list/components/film-card-item";
import {Footer} from "@components/footer/footer";

export default class MyList extends React.Component {
  render() {

    return (
      <div className="user-page">
        <Header isUserPage={true}>
          <h1 className="page-title user-page__title">My list</h1>
        </Header>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">CatalogNav</h2>

          <div className="catalog__movies-list">
            {this.props.favoriteFilms.length === 0 &&
              <div className="mx-auto">
                Список фильмов пуст :(
              </div>
            }
            {this.props.favoriteFilms.map((film, index) => {
              return <FilmCardItem key={film.id} data={film} index={index} />;
            })}
          </div>
        </section>
        <Footer/>
      </div>
    );
  }
}

MyList.propTypes = {
  favoriteFilms: propTypes.array.isRequired
};
