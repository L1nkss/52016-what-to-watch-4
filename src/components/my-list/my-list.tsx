import Header from "@components/header/header.connect";
import FilmCardItem from "@components/film-list/components/film-card-item";
import {Footer} from "@components/footer/footer";
import {TFilm} from "../../constants/types";
import * as React from "react";

interface IMyList {
  favoriteFilms: Array<TFilm>
}

const MyList = (props: IMyList) => {
  return (
    <div className="user-page">
      <Header isUserPage={true}>
        <h1 className="page-title user-page__title">My list</h1>
      </Header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">CatalogNav</h2>

        <div className="catalog__movies-list">
          {props.favoriteFilms.length === 0 &&
          <div className="mx-auto">
            Список фильмов пуст :(
          </div>
          }
          {props.favoriteFilms.map((film) => {
            return <FilmCardItem key={film.id} data={film} />;
          })}
        </div>
      </section>
      <Footer/>
    </div>
  );
};

export default MyList;
