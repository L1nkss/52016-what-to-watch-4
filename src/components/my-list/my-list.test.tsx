import * as React from "react";
import * as ShallowRenderer from 'react-test-renderer/shallow';
import {MyList} from "./my-list.connect";

const films = [{
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  previewImage: `fantastic-beasts-the-crimes-of-grindelwald`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  id: 2
}];

describe(`Testing my list component`, () => {
  it(`Component should successfully rendered`, () => {
    const myShallowRenderer = ShallowRenderer.createRenderer();
    myShallowRenderer.render(<MyList favoriteFilms={films}/>);
    const result = myShallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
