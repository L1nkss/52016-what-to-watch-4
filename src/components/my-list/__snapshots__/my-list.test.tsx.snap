// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing my list component Component should successfully rendered 1`] = `
<div
  className="user-page"
>
  <Memo(Connect(Component))
    isUserPage={true}
  >
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
  </Memo(Connect(Component))>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      CatalogNav
    </h2>
    <div
      className="catalog__movies-list"
    >
      <FilmCardItem
        data={
          Object {
            "id": 2,
            "name": "Fantastic Beasts: The Crimes of Grindelwald",
            "previewImage": "fantastic-beasts-the-crimes-of-grindelwald",
            "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          }
        }
      />
    </div>
  </section>
  <Unknown />
</div>
`;
